automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoSpec.scala

```diff
@@ -1,56 +1,157 @@
 package org.broadinstitute.dsde.workbench.leonardo
 
+import java.io.File
+import java.nio.file.Files
+import java.time.Instant
+
+import org.broadinstitute.dsde.workbench.{ResourceFile, WebBrowserSpec}
 import org.broadinstitute.dsde.workbench.config.AuthToken
 import org.broadinstitute.dsde.workbench.leonardo.ClusterStatus.ClusterStatus
+import org.broadinstitute.dsde.workbench.util.LocalFileUtil
+import org.openqa.selenium.WebDriver
 import org.scalatest.concurrent.Eventually
+import org.scalatest.selenium.WebBrowser
 import org.scalatest.time.{Minutes, Seconds, Span}
-import org.scalatest.{FreeSpec, Matchers}
+import org.scalatest.{FreeSpec, Matchers, ParallelTestExecution}
 
 import scala.util.Random
 
-class LeonardoSpec extends FreeSpec with Matchers with Eventually {
+class LeonardoSpec extends FreeSpec with Matchers with Eventually with WebBrowserSpec with WebBrowser with ParallelTestExecution with LocalFileUtil {
   implicit val ronAuthToken = AuthToken(LeonardoConfig.Users.ron)
 
+  val project = GoogleProject(LeonardoConfig.Projects.default)
+  val sa = GoogleServiceAccount(LeonardoConfig.Leonardo.notebooksServiceAccountEmail)
+  val bucket = GcsBucketName("mah-bukkit")
+
+  // ------------------------------------------------------
+
+  def clusterCheck(cluster: Cluster,
+                   requestedLabels: Map[String, String],
+                   expectedName: ClusterName,
+                   expectedStatuses: Iterable[ClusterStatus]): Unit = {
+
+    cluster.googleProject shouldBe project
+    cluster.clusterName shouldBe expectedName
+    cluster.googleServiceAccount shouldBe sa
+    cluster.googleBucket shouldBe bucket
+    cluster.labels shouldBe requestedLabels ++ DefaultLabels(expectedName, project, bucket, sa, None).toMap
+    expectedStatuses should contain (cluster.status)
+  }
+
+  // creates a cluster and checks to see that it reaches the Running state
+  def createAndMonitor(googleProject: GoogleProject, clusterName: ClusterName, clusterRequest: ClusterRequest): Cluster = {
+    val cluster = Leonardo.cluster.create(googleProject, clusterName, clusterRequest)
+    clusterCheck(cluster, clusterRequest.labels, clusterName, Seq(ClusterStatus.Creating))
+
+    // verify with get()
+    clusterCheck(Leonardo.cluster.get(googleProject, clusterName), clusterRequest.labels, clusterName, Seq(ClusterStatus.Creating))
+
+    // wait for "Running" or error (fail fast)
+    eventually {
+      clusterCheck(Leonardo.cluster.get(googleProject, clusterName), clusterRequest.labels, clusterName, Seq(ClusterStatus.Running, ClusterStatus.Error))
+    } (PatienceConfig(timeout = scaled(Span(6, Minutes)), interval = scaled(Span(20, Seconds))))
+
+    val runningCluster = Leonardo.cluster.get(googleProject, clusterName)
```

Comment:
You're getting the cluster (if the test goes right) in the `eventually` above, but then, once you've gotten the Running cluster, you're getting it again here. Can just use the running cluster you got in the `eventually` instead of getting it again?

Reply:
No, because it was eventually Running or Error.  This check is for Running only.  I can add a comment.

Reply:
oh I can just check status at that point.  Yes.